students/Dkuchan/ClassProject/PythonClassProject.py

```diff
@@ -0,0 +1,625 @@
+       # self.climb = {matchlist[i].matchnumber: matchlist[i].climb for i in matchlist if matchlist[i].teamnumber == self.number}
+
+
+def testfidelityUI():
+    print()
+    print()
+    print("You have chosen to test a matches fidelity.")
+    print("These matches are currently in the database.")
+    showmatchesUI()
+    print("Which match would you like to test?")
+
+
+def testmatchfidelity(matchnumber):
+    # Tests to see if there are 6 teams in a match teamnumber.
+    # Tests to make sure there are no repeats in teams.
+    matchcount = 0
+    teamvar = None
+    teamslist = []
+    for i in range(0, len(matchlist)):
+        if matchlist[i].matchnumber == matchnumber:
+            matchcount+=1
+            teamslist.append(int(matchlist[i].team))
+    print("There are " + len(teamslist) + "/6 logs in the database for match " + matchnumber + ".")
+    for i in teamlist:
+        checkteamnum = teamslist[i]
+        teaminmatch = 0
+        for j in teamlist:
+            if teamlist[j] == checkteamnum:
+                teaminmatch += 1
+        if teaminmatch > 1:
+            print("Team " + str(checkteamnum) + "appears " + teaminmatch + "time(s) in match " + matchnumber + ".")
+        else:
+            print("This match is complete.")
+
+# def updateteams():
+    # This function is intended to refresh team objects with the latest match data.teamnumber
+
+
+def teamcomparisonUI():
+    # this function creates a comparison of 2 teams for quick reference.
+    print()
+    print("Welcome to the Team Comparison Tool")
+    print("Which teams would you like to compare?: ")
+    print()
+
+
+def populateteamdict():
+    # this function creates a dict where the key is
+    # the team number (string) and the value is the team object.
+    for i in range(0, len(teamlist)):
+        teamdict.update({teamlist[i]: teamlist2[i]})
+
+
+def fillteamstats():
+    # this function pulls team data from the matchlist data
+    for i in teamdict:
+        teamdict[i].fillmatches()
+        teamdict[i].fillmatchwins()
+        teamdict[i].fillgearsdelivered()
+        teamdict[i].fillkpa()
+        teamdict[i].fillpilot()
+        teamdict[i].fillalliances()
+        teamdict[i].fillfieldpos()
+        teamdict[i].fillautogears()
+        teamdict[i].fillautokpa()
+        teamdict[i].fillclimb()
+
+
+def createteamlist():
+    # This very Janky function creates a list of team numbers, a matching list of team objects and calls the teamdict function
+    # I really dont like this function it is not really necessary and poorly concieved but I could not get the "elegant" way
+    # to work right and I needed something that worked in this case.
+    for i in range(0,len(matchlist)):
+        testteamnum = matchlist[i].team
+        if testteamnum not in teamlist:
+            teamlist.append(testteamnum)
+    for j in range(0,len(teamlist)):    #I THINK THIS IS SUPER JANKY BUT I COULD NOT GET A DIRECT CLASS LIST CREATION TO WORK! :(
```

Comment:
you almost never want to do this.

```
for team in teamlist:
    teamlist2.append(Team(team))
```
or
```
teamlist2 = [Team(t for t in teamlist)]
```
But what kind of objects are stored in teamlist? why not the Team objects themselves?